iss.py

```diff
@@ -117,15 +119,19 @@ def main():
         positions.append((lat, lon))
         print(positions)
 
-        (imageBlack, imageRed) = display.drawISS(positions)
-
-        # We're drawing the map in black and the ISS location and trajectory in red
-        # Swap it around if you'd like the inverse color scheme
-        epd.display(epd.getbuffer(imageBlack), epd.getbuffer(imageRed))
-        sleep(2)
-        epd.sleep()
+        # Refresh the display on the first fetch and then on every DISPLAY_REFRESH_INTERVAL fetch
+        if (DISPLAY_REFRESH_INTERVAL == 1 or len(positions) % DISPLAY_REFRESH_INTERVAL == 1):
```

Comment:
len(positions) % DISPLAY_REFRESH_INTERVAL == **1** so we refresh the display on the first data fetch. 
Needs the special case before the 'or' to refresh the display every time when DISPLAY_REFRESH_INTERVAL  = 1.

Other possibility could be:
`if (len(positions) % DISPLAY_REFRESH_INTERVAL == 0 or len(positions) == 1):`

Reply:
or 

if ((len(positions)-1) % DISPLAY_REFRESH_INTERVAL)

This should handle both cases. Basically you're going from length to index in array. Since we do the first fetch anyway, len(positions) will always be 1 or greater

or 
if ((len(positions) >= 1) && ((len(positions)-1) % DISPLAY_REFRESH_INTERVAL))

if you want to avoid the condition where positions is empty (if the request fails, for example, but we don't have any try for that yet in any case!) :-) 

But I'm fine with what you have! :-)